### DISCUSSION TITLE
Clarify "Multi-level inheritance is supported."

### DISCUSSION THREAD

**Discussion by chris-dura** [vuepress/core#436] [Q&A] [answered]
The docs mention "Multi-level inheritance is supported." in several places:
https://v2.vuepress.vuejs.org/reference/theme-api.html#extends
https://v2.vuepress.vuejs.org/guide/migration.html#extend

What does "multi-level inheritance" mean, exactly??

I initially was hoping it meant we could override individual default theme components, like we could in v1; however, the default theme no longer uses aliases, so afaict, if we want to override a single component, like `ToggleSidebarButton.vue` or `Sidebar.vue`, it means we have to override (or re-write) the parent components as well?

**Comment by meteorlxy** [comment, marked as answer]
"multi-level inheritance": theme B can extend theme A, and theme C can still extend theme B.

This is not supported in v1:

https://v1.vuepress.vuejs.org/theme/inheritance.html

![image](https://user-images.githubusercontent.com/18205362/132657963-d057d2a5-e425-455e-b308-48454619df49.png)

I'm not a native English speaker, so should we replace "multi-level inheritance" with "high-order inheritance"?

---

You should have also noticed this in v1 docs:

![image](https://user-images.githubusercontent.com/18205362/132658169-b0943e7f-4367-4087-8de4-a1436cdb06a2.png)

It's still marked as "beta" even now, which means that it's an experimental feature. You should not expect that it would be kept as is in v2, which is a breaking major version.

**Reply by chris-dura** [reply]
@meteorlxy -- Thanks, I didn't refer back to the v1 docs for that `TIP`. I think "multi-level inheritance" is correct, but it's not clear on its own, so maybe adding that TIP to the v2 docs would help, but obviously changing it to "a derived theme can also be extended".

> It's still marked as "beta" even now, which means that it's an experimental feature. You should not expect that it would be kept as is in v2, which is a breaking major version.

That's fair, it's a breaking semver version, but I feel the feature itself was/is still very valuable (there's certainly a few folks in the forums wanting the same capability). But, I don't really care if it's done the same way as in v1 or not.

IMO, there's a few groups of VuePress Theme users:

1. Users just want/need to use themes off-the-shelf, because they just need something simple.
2. Users who want/need to write a completely custom theme because they can't find a community theme that is even close to what they need.
3. Users who have found a community theme that does 95% of what they want, but just need to modify that last 5%

Obviously, I'm in group 3, and I don't necessarily believe that group 3 is the "minority". For my use case, that 5% means just replacing a few of the theme components. But, without something _like_ the v1 inheritance, I'll end up "forking" (copy+paste) a good chunk of the parent theme just to modify the 5% I actually care about... or, just write my own theme from the ground up.

So, I definitely have no expectations, just making an argument that the ability to override individual theme components (at least in the default theme) was handy for some folks.

**Reply by CxRes** [reply]
@chris-dura I am exactly in the same boat as you. I just want to make minor changes to the default theme component, and it seems I will have to copy over all the code of theme default for doing so. 

If you have any workarounds, will be highly appreciated!

**Reply by chris-dura** [reply]
@CxRes -- There was some discussion in an issue thread opened awhile ago about it. And, there was some recent activity I haven't dug into, so there may be some new workarounds that could be a solution: https://github.com/vuepress/vuepress-next/issues/104
